nodeup/pkg/model/context.go

```diff
@@ -485,12 +485,13 @@ func (c *NodeupModelContext) BuildPrivateKeyTask(ctx *fi.ModelBuilderContext, na
 func (c *NodeupModelContext) NodeName() (string, error) {
 	// This mirrors nodeutil.GetHostName
 	hostnameOverride := c.Cluster.Spec.Kubelet.HostnameOverride
+	cloudProvider := c.Cluster.Spec.Kubelet.CloudProvider
```

Comment:
You should use `c.Cluster.Spec.CloudProvider` here. This one is always set to the actual cloud provider and not to values such as `""` and `"external"`